### DISCUSSION TITLE
Problem installing Omnet++ Federate

### DISCUSSION THREAD

**Discussion by LDVictor** [eclipse/mosaic#73] [Q&A]
Hello people.

I'm trying to install Omnet++ Federate according to the documentation (https://www.eclipse.org/mosaic/docs/simulators/network_simulator_omnetpp/) but I'm facing some problems. I'm doing the installation for developers, with Omnet++ 5.5 and Inet 4.1 as recommended, but when I try to run, this error occurs:

`Error: ...tpp_federate_src/.modules/tools/.modules/tools/tools.lua:21: 

┌─────────────────────────────╖
│                             ║
│    REMOTE CODE EXECUTION    ║
│    YOU HAVE BEEN HACKED!    ║
│                             ║
╘═════════════════════════════╝

Your premake shouldn't be running this script!
Upgrade premake-core or apply this patch to it NOW:
https://github.com/premake/premake-core/commit/21c9985f0989932dbcc79d8770458393794a363f.patch

More info about this:
https://github.com/premake/premake-core/issues/1381#issuecomment-732473611
`

I have the most recent premake5 installed in my Ubuntu distro, so anyone knows how can I fix this?

**Comment by kschrab** [comment]
Thanks for noting that down! Have you tried to upgrade the used premake5 version in the omnetpp_installer.sh script?

```diff
omnet_federate_url="https://github.com/mosaic-addons/omnetpp-federate/archive/21.0.zip"
inet_src_url="https://github.com/inet-framework/inet/releases/download/v4.1.1/inet-4.1.1-src.tgz"

-premake5_url="https://github.com/premake/premake-core/releases/download/v5.0.0-alpha14/premake-5.0.0-alpha14-linux.tar.gz"
+premake5_url="https://github.com/premake/premake-core/releases/download/v5.0.0-alpha15/premake-5.0.0-alpha15-linux.tar.gz"
premake5_tar="$(basename "$premake5_url")"
premake5_autoconf_url="https://github.com/Blizzard/premake-autoconf/archive/master.zip"
premake5_autoconf_zip="$(basename "$premake5_autoconf_url")"
```

I am not sure if this is already fixed in this version though.

**Reply by LDVictor** [reply]
Thanks for responding me. I tried this upgrade that you suggested, but the same error occurs. I think it's related with premake-autoconf, because this error appears right after the downloads of premake-autoconf-master files (api.lua, autoconf.lua, clang.lua, gcc.lua, msc.lua).

`Progress: Downloading 'premake5' from: https://github.com/premake/premake-core/releases/download/v5.0.0-alpha15/premake-5.0.0-alpha15-linux.tar.gz
Progress: Downloading 'premake-autoconf' from: https://github.com/Blizzard/premake-autoconf/archive/master.zip
Progress: Extracting premake and premake-autoconf ...
premake5
Archive:  ../master.zip
d403eb6a3f3634b36258047ba9164609c2099676
   creating: premake-autoconf-master/
  inflating: premake-autoconf-master/LICENSE.txt  
  inflating: premake-autoconf-master/README.md  
  inflating: premake-autoconf-master/_manifest.lua  
  inflating: premake-autoconf-master/_preload.lua  
  inflating: premake-autoconf-master/api.lua  
  inflating: premake-autoconf-master/autoconf.lua  
  inflating: premake-autoconf-master/clang.lua  
  inflating: premake-autoconf-master/gcc.lua  
  inflating: premake-autoconf-master/msc.lua  
Progress: Building MOSAIC OMNeT++ Federate ...
Error: ...ed/omnetpp/omnetpp_federate_src/.modules/tools/tools.lua:21: 

┌─────────────────────────────╖
│                             ║
│    REMOTE CODE EXECUTION    ║
│    YOU HAVE BEEN HACKED!    ║
│                             ║
╘═════════════════════════════╝

Your premake shouldn't be running this script!
Upgrade premake-core or apply this patch to it NOW:
https://github.com/premake/premake-core/commit/21c9985f0989932dbcc79d8770458393794a363f.patch

More info about this:
https://github.com/premake/premake-core/issues/1381#issuecomment-732473611
`
